### PR TITLE
chore: clean up client-core dependencies

### PR DESCRIPTION
**Pull request by mutianf**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-bigtable-hbase/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

### CODE REVIEW

File: bigtable-client-core-parent/bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableAdmin.java [hunk 1]
```diff
@@ -119,11 +117,11 @@ public AbstractBigtableAdmin(CommonConnection connection) throws IOException {
     String clusterId =
         configuration.get(BigtableOptionsFactory.BIGTABLE_SNAPSHOT_CLUSTER_ID_KEY, null);
     if (clusterId != null) {
-      BigtableInstanceName bigtableInstanceName =
-          new BigtableInstanceName(
+      bigtableSnapshotClusterName =
```
Comment (kolea2) [inline bigtable-client-core-parent/bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableAdmin.java:120]: is this right? Are these replacements doing the same thing?

Comment (kolea2) [reply bigtable-client-core-parent/bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableAdmin.java:120]: nevermind, I see what this is doing now.

File: bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/veneer/BigtableHBaseVeneerSettings.java [hunk 2]
```diff
@@ -219,6 +219,22 @@ public long getBatchingMaxRequestSize() {
     return batchingMaxMemory;
   }
 
+  @Override
+  public String getUserAgent() {
```
Comment (igorbernstein2) [inline bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/veneer/BigtableHBaseVeneerSettings.java:223]: Maybe getEffectiveUserAgent()?
Also if this exists just for CheckConfig, I think we should probably drop CheckConfig and keep this private

File: bigtable-client-core-parent/bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableAdmin.java [hunk 3]
```diff
@@ -100,7 +98,7 @@
   private final BigtableHBaseSettings settings;
   protected final CommonConnection connection;
   protected final AdminClientWrapper adminClientWrapper;
-  private BigtableClusterName bigtableSnapshotClusterName;
+  private ClusterName bigtableSnapshotClusterName;
```
Comment (igorbernstein2) [inline bigtable-client-core-parent/bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableAdmin.java:101]: I've been wanting to drop the gapic generated resource names. Can we stick to a String here?

File: bigtable-client-core-parent/bigtable-hbase/src/test/java/com/google/cloud/bigtable/hbase/wrappers/veneer/TestAdminClientVeneerApi.java [hunk 4]
```diff
@@ -58,12 +56,13 @@
   private static final String TABLE_ID_2 = "fake-Table-id-2";
   private static final String TABLE_NAME =
       NameUtil.formatTableName(PROJECT_ID, INSTANCE_ID, TABLE_ID_1);
-  private static final BigtableInstanceName INSTANCE_NAME =
-      new BigtableInstanceName(PROJECT_ID, INSTANCE_ID);
+  private static final InstanceName INSTANCE_NAME = InstanceName.of(PROJECT_ID, INSTANCE_ID);
```
Comment (igorbernstein2) [inline bigtable-client-core-parent/bigtable-hbase/src/test/java/com/google/cloud/bigtable/hbase/wrappers/veneer/TestAdminClientVeneerApi.java:59]: I think we should stick to strings & NameUtil

File: bigtable-dataflow-parent/bigtable-hbase-beam/clirr-ignored-differences.xml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- see http://www.mojohaus.org/clirr-maven-plugin/examples/ignored-differences.html -->
+<differences>
+    <difference>
```
Comment (igorbernstein2) [inline bigtable-dataflow-parent/bigtable-hbase-beam/clirr-ignored-differences.xml:4]: please add a comment: "In the migration to v2, BigtableOptions along with bigable-client-core has been removed

File: bigtable-dataflow-parent/bigtable-hbase-beam/src/main/java/com/google/cloud/bigtable/beam/CloudBigtableIO.java [hunk 6]
```diff
@@ -791,17 +754,14 @@ public BufferedMutatorDoFn(CloudBigtableConfiguration config) {
     }
 
     @Setup
-    public synchronized void setup() {
```
Comment (igorbernstein2) [inline bigtable-dataflow-parent/bigtable-hbase-beam/src/main/java/com/google/cloud/bigtable/beam/CloudBigtableIO.java:794]: Please a TODO for re-adding this
Also we should chat about how we can re-add this

File: bigtable-dataflow-parent/bigtable-hbase-beam/src/main/java/com/google/cloud/bigtable/beam/CloudBigtableIO.java [hunk 7]
```diff
@@ -791,17 +754,14 @@ public BufferedMutatorDoFn(CloudBigtableConfiguration config) {
...
-    }
+    public synchronized void setup() {}
 
     protected BufferedMutator createBufferedMutator(Object context, String tableName)
         throws IOException {
       return getConnection()
           .getBufferedMutator(
               new BufferedMutatorParams(TableName.valueOf(tableName))
-                  .writeBufferSize(BulkOptions.BIGTABLE_MAX_MEMORY_DEFAULT)
+                  .writeBufferSize(BIGTABLE_MAX_MEMORY_DEFAULT)
```
Comment (igorbernstein2) [inline bigtable-dataflow-parent/bigtable-hbase-beam/src/main/java/com/google/cloud/bigtable/beam/CloudBigtableIO.java:764]: why this necessary?

File: bigtable-dataflow-parent/bigtable-hbase-beam/src/main/java/com/google/cloud/bigtable/beam/CloudBigtableScanConfiguration.java [hunk 8]
```diff
@@ -273,9 +273,8 @@ public CloudBigtableScanConfiguration build() {
     public ReadRowsRequest get() {
```
Comment (igorbernstein2) [inline bigtable-dataflow-parent/bigtable-hbase-beam/src/main/java/com/google/cloud/bigtable/beam/CloudBigtableScanConfiguration.java:273]: I think we should deprecate bigtable protos on the public surface (ie Builder#withRequest)

File: bigtable-dataflow-parent/bigtable-hbase-beam/src/test/java/com/google/cloud/bigtable/beam/CloudBigtableIOIntegrationTest.java [hunk 9]
```diff
@@ -27,6 +26,7 @@
 import java.util.concurrent.Executors;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicInteger;
+import java.util.logging.Logger;
```
Comment (igorbernstein2) [inline bigtable-dataflow-parent/bigtable-hbase-beam/src/test/java/com/google/cloud/bigtable/beam/CloudBigtableIOIntegrationTest.java:29]: I think this requires a bit of thought. We are really inconsistent with the loggers that are used in the project. Lets chat offline

File: bigtable-dataflow-parent/bigtable-hbase-beam/src/test/java/com/google/cloud/bigtable/beam/CloudBigtableIOTest.java [hunk 10]
```diff
@@ -130,9 +129,35 @@ public void testSampleRowKeys() throws Exception {
         new Comparator<SourceWithKeys>() {
           @Override
           public int compare(SourceWithKeys o1, SourceWithKeys o2) {
-            return ByteStringComparator.INSTANCE.compare(
-                o1.getConfiguration().getStartRowByteString(),
-                o2.getConfiguration().getStartRowByteString());
+            ByteString key1 = o1.getConfiguration().getStartRowByteString();
```
Comment (igorbernstein2) [inline bigtable-dataflow-parent/bigtable-hbase-beam/src/test/java/com/google/cloud/bigtable/beam/CloudBigtableIOTest.java:132]: I think you can use the ByteStringComparator from java-bigtable

File: bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/veneer/BigtableHBaseVeneerSettings.java (hunk 2 above)
Comment (mutianf) [reply bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/veneer/BigtableHBaseVeneerSettings.java]: Reverted the change and removed CheckConfig

File: bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/CheckConfig.java [hunk 11]
```diff
@@ -58,28 +56,19 @@ public static void main(String[] args) throws IOException, GeneralSecurityExcept
...
-
-    Credentials credentials = CredentialFactory.getCredentials(options.getCredentialOptions());
-    try {
-      System.out.println("Attempting credential refresh...");
-      credentials.refresh();
-    } catch (IOException ioe) {
-      logger.warn("Encountered errors attempting to refresh credentials.", ioe);
-      return;
-    }
+    System.out.println(String.format("User Agent: %s", settings.getUserAgent()));
```
[the remark this answers is not in the document]
Comment (mutianf) [reply bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/CheckConfig.java]: Removed this class.

File: bigtable-dataflow-parent/bigtable-hbase-beam/src/main/java/com/google/cloud/bigtable/beam/CloudBigtableScanConfiguration.java (hunk 8 above)
Comment (mutianf) [reply bigtable-dataflow-parent/bigtable-hbase-beam/src/main/java/com/google/cloud/bigtable/beam/CloudBigtableScanConfiguration.java:275]: I added `@deprecated` annotation for now, and will clean up `CloudBigtableScanConfiguration` and `TemplateUtils` in a separate pr.

File: bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/util/HBaseTracingUtilities.java [hunk 12]
```diff
@@ -1,81 +0,0 @@
...
- * <pre>
- * try {
- *   StackdriverExporter.createAndRegister(projectId);
- * } catch (Throwable e) {
- *   LOG.error("Could not register stackdriver", e);
- * }
- * </pre>
- */
-@InternalExtensionOnly
-public final class HBaseTracingUtilities {
```
[the remark this answers is not in the document]
Comment (mutianf) [reply bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/util/HBaseTracingUtilities.java]: Added the class back.

**Review by igorbernstein2 — APPROVED** [review]
lgtm

**Pull request merged**